packages/inference/src/lib/getInferenceProviderMapping.ts

```
@@ -112,7 +112,10 @@ export async function resolveProvider(
 			throw new Error("Specifying a model is required when provider is 'auto'");
 		}
 		const inferenceProviderMapping = await fetchInferenceProviderMappingForModel(modelId);
-		provider = Object.keys(inferenceProviderMapping)[0] as InferenceProvider;
+		provider = Object.keys(inferenceProviderMapping)[0] as InferenceProvider | undefined;
+	}
+	if (!provider) {
+		provider = "hf-inference";
```

"auto" no?

---

we already falling back to "auto" line 104, if provider is undefined here it means inferenceProviderMapping is empty and I'd rather raise an error instead